[Security Solution] relax metadata API permissions

joeypoon opened the pull request:
## Summary

Metadata APIs no longer require `superuser`. User only needs read access to security solution app.


### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios


### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

parkiino commented on x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts:
```diff
@@ -209,3 +222,66 @@ export const getEndpointAuthzInitialState = (): EndpointAuthz => {
+  }
+
+  const siemPermissions: EndpointPermissions = SIEM_PERMISSIONS.reduce(
+    (acc, { permission, privilege }) => {
```
would the siem privilege be set through the kibana privilege toggles as well / are they already available or do we have to add another option underneath security?

joeypoon replied:
so these map directly to the top level security all/read in kibana. `siem.crud === security.all` and `siem.show === security.read`. these already existed so I'm just passing them into our `authz` and exposing them as `canReadSecuritySolution` and `canWriteSecuritySolution`.

parkiino replied:
gotcha so as long as this is checked as 'All' it should be ok right? 
<img width="759" alt="Screen Shot 2022-10-10 at 4 45 48 PM" src="https://user-images.githubusercontent.com/56409205/194949904-58543009-c229-4183-90e0-f89ff85f75f4.png">

kevinlog commented:
@joeypoon I checked this one out and have a couple question.

I set up my Role like this:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/56395104/194144800-5caab2e7-46f0-45ff-adfe-16aac6b1f8e8.png">

Key pieces:
Host Isolation: All
Fleet: None

I generated some Alerts and went to check them using this role.  I would expect to be able to see the Agent status and the Isolate options in the flyout out below.

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/56395104/194145303-2fdad9ee-0d68-43ee-914e-37b68288d1bc.png">

I also see this in the Kibana logs:
```
[2022-10-05T15:19:09.993-04:00][ERROR][plugins.securitySolution.metadata] EndpointError: User does not have adequate permissions to access Fleet agents.
    at EndpointMetadataService.getFleetAgent (/Users/kevinlogan/kibana/x-pack/plugins/security_solution/server/endpoint/services/metadata/endpoint_metadata_service.ts:317:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at EndpointMetadataService.getEnrichedHostMetadata (/Users/kevinlogan/kibana/x-pack/plugins/security_solution/server/endpoint/services/metadata/endpoint_metadata_service.ts:169:20)
    at /Users/kevinlogan/kibana/x-pack/plugins/security_solution/server/endpoint/routes/metadata/handlers.ts:143:15
    at Router.handle (/Users/kevinlogan/kibana/node_modules/@kbn/core-http-router-server-internal/target_node/src/router.js:163:30)
    at handler (/Users/kevinlogan/kibana/node_modules/@kbn/core-http-router-server-internal/target_node/src/router.js:124:50)
    at exports.Manager.execute (/Users/kevinlogan/kibana/node_modules/@hapi/hapi/lib/toolkit.js:60:28)
    at Object.internals.handler (/Users/kevinlogan/kibana/node_modules/@hapi/hapi/lib/handler.js:46:20)
    at exports.execute (/Users/kevinlogan/kibana/node_modules/@hapi/hapi/lib/handler.js:31:20)
    at Request._lifecycle (/Users/kevinlogan/kibana/node_modules/@hapi/hapi/lib/request.js:371:32)
```

Do we still need to make UI changes to see this data in the UI?

joeypoon commented:
@kevinlog thanks for finding that. fixed in new commit.
![Screen Shot 2022-10-05 at 6 26 32 PM](https://user-images.githubusercontent.com/11009772/194183485-97bd8694-9fa5-4125-a62a-ae5cfaa2ceeb.png)

also, I added `ci:cloud-deploy` so it's easier to test.

kevinlog commented:
@joeypoon tried it again, looks better!  But I see one more issue.

With the same settings as I had above:
Security: ALL
Host Isolation: ALL
All other sub-features: NONE
Fleet: NONE

I can now see Agent details and the access the isolate option from the Alert flyout ✅ 
<img width="1724" alt="image" src="https://user-images.githubusercontent.com/56395104/194309388-099bb4b1-3687-4224-9411-a16af215e6c1.png">

I can complete the isolate action from the flyout ✅ 
<img width="1724" alt="image" src="https://user-images.githubusercontent.com/56395104/194309532-3ce80492-65fe-44a3-aded-e51d652f2f4b.png">

However, I don't think the actual Action document is being submitted to the relevant indices ❌ 
I never see the Agent status update and my Endpoint VM isn't being isolated.
When I switch over to a `superuser` role to check the Actions log, I can see that there is nothing that shows up in the log.  

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/56395104/194311256-fb5901a2-e9fe-4b10-87ca-a4415c433a55.png">

My guess is that we're not using `asInternalUser` somewhere to successfully write the document to `.logs-endpoint.actions-*` and `.fleet-actions-*`.  Could you check the `isolate` and `unisolate` routes again?  Specifically, where we submit the action documents.

joeypoon commented:
@kevinlog you were right, response actions wasn't writing `agentId` into the doc so while the doc was written, it wasn't associated to the agent correctly. fixed in latest commit.

kevinlog commented:
@joeypoon tried it again, looks great!

I can see the correct data and do Host Isolations.  The actions propagate correctly!
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/56395104/194386534-744f1cb8-26b6-4cd8-9857-e75ad66f9122.png">

In addition, the Agent data appears correctly in the Hosts view
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/56395104/194387329-ce41761a-743a-4fb6-91ec-3bb1bda1d158.png">

parkiino reviewed (approved):
checked it out. looks great!
